### ISSUE TITLE
No way to login. /tmp/.mount_.../resources/app.asar.unpacked/build/bin/linux/legendary auth --code <redacted>

### ISSUE THREAD

**Issue by RafaelLinux** [Heroic-Games-Launcher/HeroicGamesLauncher#2854]
### Describe the bug

I tried every new release of Heroic since 6 months ago and finally I write here to get help

- I can login to my EPIC account without issues (my default browser is Firefox)
- I'm using openSUSE Tumbleweed and latest (12/7/2022) downloaded AppImage version
- I can launch Heroic.Appimage without issues

```
DevTools listening on ws://127.0.0.1:9222/devtools/browser/399aa150-0109-4ce3-beec-0855ff5dd57d
(14:57:02) INFO:    [Legendary]:        Refreshing library...
(14:57:02) INFO:    [Connection]:       Connectivity: check-online
(14:57:02) INFO:    [Connection]:       Pinging external endpoints
(14:57:02) INFO:    [Legendary]:        Running command: /tmp/.mount_HeroicMZEM20/resources/app.asar.unpacked/build/bin/linux/legendary --version
(14:57:02) INFO:    [Legendary]:        Legendary location: /tmp/.mount_HeroicMZEM20/resources/app.asar.unpacked/build/bin/linux/legendary
(14:57:02) INFO:    [Gog]:              GOGDL location: /tmp/.mount_HeroicMZEM20/resources/app.asar.unpacked/build/bin/linux/gogdl
(14:57:02) INFO:    [Gog]:              GOGDL location: /tmp/.mount_HeroicMZEM20/resources/app.asar.unpacked/build/bin/linux/gogdl
Checking for update
[9638:0712/145703.017471:ERROR:vaapi_wrapper.cc(1315)] vaQuerySurfaceAttributes failed, VA error: invalid parameter
[9638:0712/145703.017581:ERROR:vaapi_wrapper.cc(1262)] FillProfileInfo_Locked failed for va_profile VAProfileH264Main and entrypoint VAEntrypointVLD
[9638:0712/145703.017637:ERROR:vaapi_wrapper.cc(1315)] vaQuerySurfaceAttributes failed, VA error: invalid parameter
[9638:0712/145703.017676:ERROR:vaapi_wrapper.cc(1262)] FillProfileInfo_Locked failed for va_profile VAProfileH264High and entrypoint VAEntrypointVLD
(14:57:04) WARNING: [Backend]:          Failed to register protocol with OS.
(14:57:04) WARNING: [Backend]:          Could not determine System Info
(14:57:04) INFO:    [Connection]:       Connectivity: online
(14:57:04) INFO:    [Backend]:          User Not Found, removing it from Store
[9601:0712/145704.471867:ERROR:nss_util.cc(357)] After loading Root Certs, loaded==false: NSS error code: -8018
(14:57:04) INFO:    [Backend]:          AreWeAntiCheatYet data downloaded
(14:57:04) INFO:    [Backend]:          Loading Screen Ready
(14:57:04) INFO:    [Backend]:          Frontend Ready
(14:57:05) INFO:    [Backend]:          Checking for new Heroic Updates
(14:57:05) INFO:    [Backend]:          Checking for current version changelog
(14:57:05) INFO:    [Winetricks]:       Downloading Winetricks
Update for version 2.8.0 is not available (latest version: 2.8.0, downgrade is disallowed).
(14:57:09) INFO:    [Backend]:          Starting the Download Queue

```
However, there is no way to get Heroic to be logged with any of two options available ("Init session ..." nor "Alternative method") even being or not logged in my EPIC thru my browser.

### Add logs

```text
(14:46:35) INFO:    [Legendary]:        Refreshing library...
(14:46:35) INFO:    [Connection]:       Connectivity: check-online
(14:46:35) INFO:    [Connection]:       Pinging external endpoints
(14:46:35) INFO:    [Legendary]:        Running command: /tmp/.mount_HeroicbvYCcE/resources/app.asar.unpacked/build/bin/linux/legendary --version
(14:46:35) INFO:    [Legendary]:        Legendary location: /tmp/.mount_HeroicbvYCcE/resources/app.asar.unpacked/build/bin/linux/legendary
(14:46:35) INFO:    [Gog]:              GOGDL location: /tmp/.mount_HeroicbvYCcE/resources/app.asar.unpacked/build/bin/linux/gogdl
(14:46:35) INFO:    [Gog]:              GOGDL location: /tmp/.mount_HeroicbvYCcE/resources/app.asar.unpacked/build/bin/linux/gogdl
(14:46:39) WARNING: [Backend]:          Failed to register protocol with OS.
(14:46:39) WARNING: [DXVKInstaller]:    App offline, skipping possible DXVK update.
(14:46:39) WARNING: [Backend]:          Could not determine System Info
(14:46:39) INFO:    [Connection]:       Connectivity: online
(14:46:39) INFO:    [Backend]:          User Not Found, removing it from Store
(14:46:39) INFO:    [Backend]:          AreWeAntiCheatYet data downloaded
(14:46:40) INFO:    [Backend]:          Loading Screen Ready
(14:46:40) INFO:    [Backend]:          Frontend Ready
(14:46:40) INFO:    [Backend]:          Checking for new Heroic Updates
(14:46:40) INFO:    [Backend]:          Checking for current version changelog
(14:46:45) INFO:    [Backend]:          Starting the Download Queue
(14:47:26) INFO:    [Legendary]:        Logging in: /tmp/.mount_HeroicbvYCcE/resources/app.asar.unpacked/build/bin/linux/legendary auth --code <redacted>
(14:47:26) INFO:    [Frontend]:         Refreshing legendary Library
(14:47:26) INFO:    [Legendary]:        Refreshing library...
(14:47:57) INFO:    [Frontend]:         Called Epic Login
(14:47:57) INFO:    [Legendary]:        Logging in: /tmp/.mount_HeroicbvYCcE/resources/app.asar.unpacked/build/bin/linux/legendary auth --code <redacted>
(14:47:57) INFO:    [Frontend]:         Refreshing legendary Library
(14:47:57) INFO:    [Legendary]:        Refreshing library...
(14:51:16) INFO:    [Legendary]:        Running command: /tmp/.mount_HeroicbvYCcE/resources/app.asar.unpacked/build/bin/linux/legendary --version
(14:51:16) INFO:    [Gog]:              Running command: /tmp/.mount_HeroicbvYCcE/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/user/.config/heroic/gog_store/auth.json --version
(14:51:33) INFO:    [Legendary]:        Getting EOS Overlay install size: /tmp/.mount_HeroicbvYCcE/resources/app.asar.unpacked/build/bin/linux/legendary eos-overlay install --path /home/user/.config/heroic/tools/eos_overlay
(14:51:33) INFO:    [Backend]:          Preventing machine to sleep
(14:51:33) INFO:    [Legendary]:        Installing EOS Overlay: /tmp/.mount_HeroicbvYCcE/resources/app.asar.unpacked/build/bin/linux/legendary -y eos-overlay install --path /home/user/.config/heroic/tools/eos_overlay
(14:51:33) INFO:    [Frontend]:         Refreshing legendary Library
(14:51:33) INFO:    [Legendary]:        Refreshing library...
(14:51:36) INFO:    [Legendary]:        Getting EOS Overlay install size: /tmp/.mount_HeroicbvYCcE/resources/app.asar.unpacked/build/bin/linux/legendary eos-overlay install --path /home/user/.config/heroic/tools/eos_overlay
(14:51:36) INFO:    [Backend]:          Preventing machine to sleep
(14:51:36) INFO:    [Legendary]:        Installing EOS Overlay: /tmp/.mount_HeroicbvYCcE/resources/app.asar.unpacked/build/bin/linux/legendary -y eos-overlay install --path /home/user/.config/heroic/tools/eos_overlay
(14:51:37) INFO:    [Frontend]:         Refreshing legendary Library
(14:51:37) INFO:    [Legendary]:        Refreshing library...
(14:51:37) INFO:    [Backend]:          Stopping Power Saver Blocker
```

### Steps to reproduce

1. Click on 'EPIC login' or 'Alternative method to login ...'

Result: Heroic seems to load a page internally, and then shows again login page. 

### Expected behavior

To get to login thru Heroic

### Heroic Version

Latest Stable

### System Information

Operating System: openSUSE Tumbleweed 20230709
KDE Plasma Version: 5.27.6
KDE Frameworks Version: 5.108.0
Qt Version: 5.15.10
Kernel Version: 6.3.9-1-default (64-bit)
Graphics Platform: X11
Processors: 4 × Intel® Core™ i5-6600K CPU @ 3.50GHz
Memory: 31.3 GiB of RAM
Graphics Processor: NVIDIA GeForce GTX 1050 Ti/PCIe/SSE2

**Comment by shadowoliver** [comment]
now epic games(a china communist party's tencent's company)use a MITM cdn "cloudflare" and may be it can not be solved.
because that's tencent and cloudflare

**Comment by shadowoliver** [comment]
because heroic use electron to view the epic games.it's blocked!!!

**Comment by shadowoliver** [comment]
previously logged in users donot have issue because they already have cookies...

**Comment by RafaelLinux** [comment]
Update: Same for  GOG login. Works thru web browsers, fails from Heroic.  :(

**Comment by flavioislima** [comment]
Just tested here Epic and GOG and had no issues logging in.
We had issues in the past with Tumbleweed and we do not officially support it. Perhaps try to get support on our Discord server to see if someone else had this issue.

**Comment by RafaelLinux** [comment]
I asked to another openSUSE Thunderbird users if they have issues with login. They told me they can log in EPIC and GOG with AppImage and FlatPak versions. So I tried again and same results: Back to login option in Heroic. However, this time I captured output from console after trying to login my GOG accout:

```
(09:31:00) INFO:    [Legendary]:        Refreshing library...
(09:31:00) INFO:    [Connection]:       Connectivity: check-online
(09:31:00) INFO:    [Connection]:       Pinging external endpoints
(09:31:00) INFO:    [Legendary]:        Running command: /tmp/.mount_HeroicwWCXVe/resources/app.asar.unpacked/build/bin/linux/legendary --version
(09:31:01) INFO:    [Legendary]:        Legendary location: /tmp/.mount_HeroicwWCXVe/resources/app.asar.unpacked/build/bin/linux/legendary
(09:31:01) INFO:    [Gog]:              GOGDL location: /tmp/.mount_HeroicwWCXVe/resources/app.asar.unpacked/build/bin/linux/gogdl
(09:31:01) INFO:    [Gog]:              GOGDL location: /tmp/.mount_HeroicwWCXVe/resources/app.asar.unpacked/build/bin/linux/gogdl
Checking for update
libva error: vaGetDriverNameByIndex() failed with unknown libva error, driver_name = (null)
(09:31:02) WARNING: [Backend]:          Failed to register protocol with OS.
(09:31:02) WARNING: [Backend]:          Could not determine System Info
(09:31:02) INFO:    [Connection]:       Connectivity: online
(09:31:02) INFO:    [Backend]:          User Not Found, removing it from Store
[31447:0719/093102.621090:ERROR:nss_util.cc(357)] After loading Root Certs, loaded==false: NSS error code: -8018
(09:31:02) INFO:    [Backend]:          AreWeAntiCheatYet data downloaded
(09:31:03) INFO:    [Backend]:          Loading Screen Ready
(09:31:03) INFO:    [Backend]:          Frontend Ready
(09:31:03) INFO:    [Backend]:          Checking for new Heroic Updates
(09:31:03) INFO:    [Backend]:          Checking for current version changelog
(09:31:03) INFO:    [Winetricks]:       Downloading Winetricks
Update for version 2.8.0 is not available (latest version: 2.8.0, downgrade is disallowed).
(09:31:08) INFO:    [Backend]:          Starting the Download Queue
(09:38:45) INFO:    [Gog]:              Logging using GOG credentials
(09:38:45) INFO:    [Gog]:              Running command: /tmp/.mount_HeroicwWCXVe/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/falele/.config/heroic/gog_store/auth.json auth --code <redacted>
(09:38:45) ERROR:   [Gog]:              GOG login failed to parse std output from gogdl. stdout: 

```

About "libva" error, appears now cause I just updated kernel yesterday and need to recompile nvidia-vaapi-driver. 

Ask me for any other neccesary info about.

Thanks

**Comment by RafaelLinux** [comment]
I tried yesterday update. No way still. Only distinct thing is when login to GOG, or it seems is distinct to me. But no way to login anyway.

**Comment by RafaelLinux** [comment]
Despite it's closed, I want to note here that with 2.10.0 version all is finally working fine!!!
Thank you!!

**Comment by flavioislima** [comment]
> Despite it's closed, I want to note here that with 2.10.0 version all is finally working fine!!! Thank you!!

Nice! Glad to hear!
